### PR TITLE
🐛 bug: Fix cookie secure flag with SameSite None

### PR DESCRIPTION
**Pull request by gaby**
## Summary
- ensure secure cookies when using `SameSite=None`
- add test verifying secure flag is auto-enabled for `SameSite=None`

Mozilla Docs: https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Set-Cookie?utm_source=chatgpt.com#none

Chrome Docs: https://developers.google.com/search/blog/2020/01/get-ready-for-new-samesitenone-secure

### CODE REVIEW

File: ctx.go [hunk 1]
```diff
@@ -425,6 +425,10 @@ func (c *DefaultCtx) Cookie(cookie *Cookie) {
 		cookie.Path = "/"
 	}
 
+	if utils.ToLower(cookie.SameSite) == CookieSameSiteNoneMode && !cookie.Secure {
```
Comment (sixcolors) [inline ctx.go:428]: To avoid allocations, you should use utils.EqualFold instead.

Comment (sixcolors) [reply ctx.go:428]: Also related #3608

Comment (gaby) [reply ctx.go:428]: @sixcolors I'm closing this PR then.
